src/Bicep.Core/SemanticModel/FileSymbol.cs

```diff
@@ -61,14 +61,28 @@ public override IEnumerable<ErrorDiagnostic> GetDiagnostics()
                 .Where(decl => decl.NameSyntax.IsValid)
                 .GroupBy(decl => decl.Name)
                 .Where(group => group.Count() > 1);
-            
+
             foreach (IGrouping<string, DeclaredSymbol> group in duplicateSymbols)
             {
                 foreach (DeclaredSymbol duplicatedSymbol in group)
                 {
                     yield return this.CreateError(duplicatedSymbol.NameSyntax, b => b.IdentifierMultipleDeclarations(duplicatedSymbol.Name));
                 }
             }
+
+            var reservedSymbols = this.AllDeclarations
+                .Where(decl => decl.NameSyntax.IsValid)
+                .Where(decl => this.ImportedNamespaces.Select(ns => ns.Name).Contains(decl.Name))
+                .ToList();
+
+            foreach (DeclaredSymbol reservedSymbol in reservedSymbols)
```

Comment:
Checking for reserved namespaces after duplicate symbol, not sure if order matter here.

Reply:
The effect of that will be that if you declare 2 identifiers called `az` they get 2 errors each: one about the duplication and one about the identifier being reserved. I think that's fine.